rclpy/src/rclpy/qos.hpp

```diff
@@ -0,0 +1,75 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLPY__QOS_HPP_
+#define RCLPY__QOS_HPP_
+
+#include <rmw/qos_profiles.h>
+
+#include <pybind11/pybind11.h>
+
+#include <string>
+
+
+namespace py = pybind11;
+
+namespace rclpy
+{
+
+/// Result type for checking QoS compatibility
+/**
+ * \see rclpy::qos_check_compatible()
+ */
+struct QoSCheckCompatibleResult
+{
+  /// Compatibility result.
+  rmw_qos_compatibility_type_t compatibility;
+
+  /// Reason for a (possible) incompatibility.
+  /**
+   * Set if compatiblity is RMW_QOS_COMPATIBILITY_WARNING or RMW_QOS_COMPATIBILITY_ERROR.
+   * Not set if the QoS profiles are compatible.
+   */
+  std::string reason;
```

Comment:
Nit, the above could be simplify this to `char reason[2048];` and ditch `reason_c+str` in the function.

Reply:
I think you can ditch `reason_size` too with `sizeof(QoSCheckCompatibleResult::reason)` to avoid duplicating the number  `2048`.